atomic_reactor/plugins/post_koji_upload.py

```diff
@@ -0,0 +1,642 @@
+                    'filesize': os.path.getsize(path),
+                    'checksum': checksums['md5sum'],
+                    'checksum_type': 'md5'}
+
+        return metadata
+
+    def get_builder_image_id(self):
+        """
+        Find out the docker ID of the buildroot image we are in.
+        """
+
+        try:
+            buildroot_tag = os.environ["OPENSHIFT_CUSTOM_BUILD_BASE_IMAGE"]
+        except KeyError:
+            return ''
+
+        try:
+            pod = self.osbs.get_pod_for_build(self.build_id)
+            all_images = pod.get_container_image_ids()
+        except OsbsException as ex:
+            self.log.error("unable to find image id: %r", ex)
+            return buildroot_tag
+
+        try:
+            return all_images[buildroot_tag]
+        except KeyError:
+            self.log.error("Unable to determine buildroot image ID for %s",
+                           buildroot_tag)
+            return buildroot_tag
+
+    def get_buildroot(self, build_id):
+        """
+        Build the buildroot entry of the metadata.
+
+        :return: dict, partial metadata
+        """
+
+        docker_info = self.tasker.get_info()
+        host_arch, docker_version = get_docker_architecture(self.tasker)
+
+        buildroot = {
+            'id': 1,
+            'host': {
+                'os': docker_info['OperatingSystem'],
+                'arch': host_arch,
+            },
+            'content_generator': {
+                'name': PROG,
+                'version': atomic_reactor_version,
+            },
+            'container': {
+                'type': 'docker',
+                'arch': os.uname()[4],
+            },
+            'tools': [
+                {
+                    'name': tool['name'],
+                    'version': tool['version'],
+                }
+                for tool in get_version_of_tools()] + [
+                {
+                    'name': 'docker',
+                    'version': docker_version,
+                },
+            ],
+            'components': self.get_rpms(),
+            'extra': {
+                'osbs': {
+                    'build_id': build_id,
+                    'builder_image_id': self.get_builder_image_id(),
+                }
+            },
+        }
+
+        return buildroot
+
+    def get_logs(self):
+        """
+        Build the logs entry for the metadata 'output' section
+
+        :return: list, Output instances
+        """
+
+        output = []
+
+        # Collect logs from server
+        try:
+            logs = self.osbs.get_build_logs(self.build_id)
+        except OsbsException as ex:
+            self.log.error("unable to get build logs: %r", ex)
+        else:
+            # Deleted once closed
+            logfile = NamedTemporaryFile(prefix=self.build_id,
+                                         suffix=".log",
+                                         mode='wb')
+            logfile.write(logs.encode('UTF-8'))
```

Comment:
This needs updating to match the recent change to koji_promote: https://github.com/projectatomic/atomic-reactor/commit/9e09d607d1c5b648fbbc92224ec300b930d0945e